### PR TITLE
Editor: Unify the preview dropdown between post and site editors

### PR DESCRIPTION
**Pull request by youknowriad**
Related #52632 

## What?

This PR continues the work on the great unification between post and site editors. This PR unifies the DevicePreview or PreviewDropdown component. It also brings some consistency meaning that now:

 - When you're editing a page, either it's in the site or post editors, it offers you a way to preview the page in the frontend.
 - When you're editing a template it shows the "view site" link instead.

## Testing instructions

 - Check the device preview dropdown in both post and site editors.

### CODE REVIEW

File: packages/editor/src/components/preview-dropdown/index.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+					</MenuGroup>
+					{ isTemplate && (
+						<MenuGroup>
+							<MenuItem
+								href={ homeUrl }
+								target="_blank"
+								icon={ external }
+								onClick={ onClose }
+							>
+								{ __( 'View site' ) }
```
Comment (youknowriad) [inline packages/editor/src/components/preview-dropdown/index.js:106]: @jasmussen @jameskoster IMO, now that we have a "view site" link in the sidebar of the site editor, this "view site" doesn't make sense anymore and should be removed. I'm keeping for now but let me know if you agree and whether I should just get rid of it.

File: packages/block-editor/src/components/preview-options/README.md [hunk 2]
```diff
@@ -1,94 +0,0 @@
...
-## Development guidelines
-
-### Usage
-
-Renders the previews options of the editor in a dropdown menu.
-
-```jsx
-import { Icon, MenuGroup } from '@wordpress/components';
-import { PostPreviewButton } from '@wordpress/editor';
-import { __experimentalPreviewOptions as PreviewOptions } from '@wordpress/block-editor';
```
Comment (t-hamano) [inline packages/block-editor/src/components/preview-options/README.md:23]: Although this component has a `__experimental` prefix, I am concerned that this PR may impact consumers using this component. My understanding is that once an API is exposed to the core, it should remain backward compatible regardless of the prefix, what do you think?

However, [this component does not seem to be used](https://github.com/search?q=org%3Awp-plugins%20__experimentalPreviewOptions&type=code), at least for the plugins published in the plugin directory.

Comment (youknowriad) [reply packages/block-editor/src/components/preview-options/README.md:23]: Yes, I had already checked that it was not used. And for experimental and stable ones, they were introduced with the intent that they can be changed, so there's an exception where we can assess for each one of these whether it makes sense to keep them or not. I chose to keep it but deprecate it without rendering anything, that way it's not an error that would produce a white page or something serious but a component that won't be rendered. I also don't really expect third-parties to reach for this component to be honest.

Comment (t-hamano) [reply packages/block-editor/src/components/preview-options/README.md:23]: Oh, I missed that deprecation was [implemented](https://github.com/WordPress/gutenberg/pull/56921/files#diff-61a1980577fb29dd7a7ed4d0a4c23919d5c6bbae93517a9957f80e1c04d50392R6-R11) and it was returning empty. I also don't think there is a case where third-party developers use this component, and as long as it does not return a critical error, I think there is no problem.

**Review by t-hamano — COMMENTED** [review]
Overall, looks good to me! There are only two points of concern.

In the Post Editor, I noticed that the button tooltip has changed from "Preview" to "View". However, this does match the context of the dropdown menu's aria label, so I consider it an improvement.

view-tooltip

There are no longer disabled button in template parts and patterns. In the past, we intentionally left this disabled button. Here is the background: https://github.com/WordPress/gutenberg/pull/53913#issuecomment-1697346079

template-disabled-button

File: packages/editor/src/components/preview-dropdown/index.js (hunk 1 above)
Comment (jasmussen) [reply packages/editor/src/components/preview-dropdown/index.js:106]: Can you clarifyt which "View site" link in the sidebar you're referring to? I'm only seeing this:

Screenshot 2023-12-11 at 08 20 32

Do you mean this one?

Screenshot 2023-12-11 at 08 22 07

I think there's an opportunity for furthering the two modes, site view and edit view, a bit more before we remove the link from the dropdown. Hopefully we can consolidate in the future, but for now this feels an ergonomic location for the link when you're in edit view.

Comment (youknowriad) [reply packages/editor/src/components/preview-dropdown/index.js:106]: Ok, I'll keep it for now.

**Comment by youknowriad** [comment]
@t-hamano Both of these were intended in this PR. Basically, each time I find a problem, I try to lean on the "simple" approach first.

 - For the label, it doesn't really make sense to me to have a different label for the same UI in site and post editors
 - For the "disabled" state, I think ideally, it shouldn't be disabled, in the case of a "resizable" editor, clicking "mobile" could automatically resize to the mobile size... That said, it's out of scope, and I'm fine adding a disabled prop for now.

**Comment by youknowriad** [comment]
Ok, I've restored the disabled state.

**Review by t-hamano — APPROVED** [review]
I think we need to fix the E2E test, but it looks good to me.

**Pull request merged**